src/reporters/reporter-util.ts

```diff
@@ -0,0 +1,209 @@
+/**
+ * MIT License
+ *
+ * Copyright (c) 2020-present, Elastic NV
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+import { fileURLToPath } from 'url';
+import { resolve, sep } from 'path';
+import { readFileSync } from 'fs';
+import { inspect } from 'util';
+import { gray } from 'kleur/colors';
+import { codeFrameColumns } from '@babel/code-frame';
+import StackUtils from 'stack-utils';
+import { Location, StackFrame, TestError } from '../common_types';
+
+const stackUtils = new StackUtils({ internals: StackUtils.nodeInternals() });
+
+const SYNTHETICS_PATH = resolve(__dirname, '..', '..');
+const PW_CORE_PATH = require.resolve('playwright-core');
+
+function prepareStackFrame(line: string): StackFrame {
+  const frame = stackUtils.parseLine(line);
+  if (!frame) {
+    return;
+  }
+  // ignore node internals
+  if (frame.file?.startsWith('internal') || frame.file?.startsWith('node:')) {
+    return;
+  }
+
+  // handle relative URLs
+  let file = frame.file?.startsWith('file://')
```

Comment:
Question partially related to the [comment](https://github.com/elastic/synthetics/pull/826/files#r1339914812) below. If my assumption there was correct (using sep for ignoring node_modules in Window OS, too)

Would it make sense then to think that maybe "file://" will be different in Windows too? If so, do we want to handle that case?